hack/tools/go.mod

```diff
@@ -1,12 +1,14 @@
 module github.com/metal3-io/cluster-api-provider-metal3/hack/tools
 
-go 1.22
+go 1.22.0
```

Comment:
I think we need to set this to `1.22.3` directly, and then it will not inject the toolchain directive.

Reply:
can we hold this, I want to check manually if we can still lvie without pinning the patch version

Reply:
/hold

Reply:
I did check it in Ironic operator already, and it will inject either the toolchain or the patch version as it is required by the module.

Reply:
Yeah, we've had this discussion recently. Unfortunately, now we need to live with patch versions.